versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5967 @@
+- `ceil`: Rounds **up** to the next higher integer.
+- `round`: Rounds to the nearest integer based on standard rounding rules ("round half up").
+
+**Parameters**:
+
+1. `Float`: the number to round.
+
+**Returns**: An integer.
+
+**Example**
+
+```wdl
+# all these expressions evaluate to true
+Boolean all_true = [
+  floor(1.0) == 1,
+  floor(1.9) == 1,
+  ceil(2.0) == 2,
+  ceil(2.1) == 3,
+  round(1.49) == 1,
+  round(1.50) == 2
+]
+```
+
+## ✨ Int min(Int, Int), Float min(Float, Float), Float min(Int, Float), Float min(Float, Int)
+
+Returns the smaller of two values. If both values are `Int`s, the return value is an `Int`, otherwise it is a `Float`.
+
+**Parameters**:
+
+1. `Int|Float`: the first number to compare.
+2. `Int|Float`: the second number to compare.
+
+**Returns**: The smaller of the two arguments.
+
+**Example**
+
+```wdl
+workflow min_test {
+  input {
+    Int value1
+    Float value2
+  }
+  output {
+    # these two expressions are equivalent
+    Float min1 = if value1 < value2 then value1 else value2
+    Float min2 = min(value1, value2)
+  }
+}
+``` 
+
+## ✨ Int max(Int, Int), Float max(Float, Float), Float max(Int, Float), Float max(Float, Int)
+
+Returns the larger of two values. If both values are `Int`s, the return value is an `Int`, otherwise it is a `Float`.
+
+**Parameters**:
+
+1. `Int|Float`: the first number to compare.
+2. `Int|Float`: the second number to compare.
+
+**Returns**: The larger of the two arguments.
+
+**Example**
+
+```wdl
+workflow max_test {
+  input {
+    Int value1
+    Float value2
+  }
+  output {
+    # these two expressions are equivalent
+    Float max1 = if value1 > value2 then value1 else value2
+    Float max2 = max(value1, value2)
+  }
+}
+``` 
+
+## String sub(String, String, String)
+
+Given 3 String parameters `input`, `pattern`, `replace`, this function replaces all non-overlapping occurrences of `pattern` in `input` by `replace`. `pattern` is a [regular expression](https://en.wikipedia.org/wiki/Regular_expression) that will be evaluated as a [POSIX Extended Regular Expression (ERE)](https://en.wikipedia.org/wiki/Regular_expression#POSIX_basic_and_extended).
+
+🗑 The option for execution engines to allow other regular expression grammars besides POSIX ERE is deprecated.
+
+**Parameters**:
+
+1. `String`: the input string.
+2. `String`: the pattern to search for.
+3. `String`: the replacement string.
+
+**Returns**: the input string, with all occurrences of the pattern replaced by the replacement string.
+
+**Example**
+
+```wdl
+String chocolike = "I like chocolate when it's late"
+
+String chocolove = sub(chocolike, "like", "love") # I love chocolate when it's late
+String chocoearly = sub(chocolike, "late", "early") # I like chocoearly when it's early
+String chocolate = sub(chocolike, "late$", "early") # I like chocolate when it's early
+String chocoearlylate = sub(chocolike, "[^ ]late", "early") # I like chocoearly when it's late
```

Comment:
@DavyCats mentioned in the regex PR that `^[ ]late` matches `olate` not just `late` so the resulting string should be `I like chocearly when it's late`

Reply:
good catch